File: Python/programmers/etc/level1/같은_숫자는_싫어.py
def solution(arr):
    answer = []
    for a in arr:
        if answer[-1:] != [a]:
            answer.append(a)
    return answer


arr = [1,1,3,3,0,1,1]
print(solution(arr)) # [1,3,0,1]

arr = [4,4,4,3,3]
print(solution(arr)) # [4,3]

File: Java/programmers/etc/level1/같은숫자는싫어.java
package CodeTest.Java.programmers.etc.level1;

import java.util.*;

public class 같은숫자는싫어 {
    public Stack<Integer> solution(int[] arr) {
        Stack<Integer> stk = new Stack<>();
        stk.push(arr[0]);
        
        for (int i=1; i<arr.length; i++) {
            if (stk.peek() != arr[i]) stk.push(arr[i]);
        }
        
        return stk;
    }
}
